databricks/koalas/groupby.py

```diff
@@ -2177,7 +2177,10 @@ def _resolve_grouping(kdf: DataFrame, by: List[Union[Series, Tuple[str, ...]]])
         new_by_series = []
         for col_or_s in by:
             if isinstance(col_or_s, Series):
-                new_by_series.append(col_or_s)
+                if col_or_s._kdf is kdf:
+                    new_by_series.append(col_or_s)
+                else:
+                    new_by_series.append(col_or_s.rename(col_or_s.name))
```

Comment:
Ah, okay. So now it can contain the same Spark column but Koalas' column labels could be different .. shell we leave a short comment here?

Reply:
maybe we should use more explicit name. let me update with a comment.